sormas-api/src/main/java/de/symeda/sormas/api/report/AggregatedCaseCountDto.java

```diff
@@ -291,39 +237,181 @@ public void setEpiWeek(Integer epiWeek) {
 		this.epiWeek = epiWeek;
 	}
 
-	@Override
-	public int hashCode() {
+	public UserReferenceDto getReportingUser() {
+		return reportingUser;
+	}
 
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + (int) deaths;
-		result = prime * result + ((disease == null) ? 0 : disease.hashCode());
-		result = prime * result + (int) labConfirmations;
-		result = prime * result + (int) newCases;
-		result = prime * result + ((ageGroup == null) ? 0 : ageGroup.hashCode());
-		return result;
+	public void setReportingUser(UserReferenceDto reportingUser) {
+		this.reportingUser = reportingUser;
 	}
 
-	@Override
-	public boolean equals(Object obj) {
+	public Date getChangeDate() {
+		return changeDate;
+	}
 
-		if (this == obj)
+	public void setChangeDate(Date changeDate) {
+		this.changeDate = changeDate;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o)
 			return true;
-		if (obj == null)
-			return false;
-		if (getClass() != obj.getClass())
-			return false;
-		AggregatedCaseCountDto other = (AggregatedCaseCountDto) obj;
-		if (deaths != other.deaths)
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (disease != other.disease)
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+		return year == that.year
+			&& epiWeek == that.epiWeek
+			&& newCases == that.newCases
+			&& labConfirmations == that.labConfirmations
+			&& deaths == that.deaths
+			&& Objects.equals(regionName, that.regionName)
+			&& Objects.equals(regionId, that.regionId)
+			&& Objects.equals(districtName, that.districtName)
+			&& Objects.equals(districtId, that.districtId)
+			&& Objects.equals(healthFacilityName, that.healthFacilityName)
+			&& Objects.equals(healthFacilityId, that.healthFacilityId)
+			&& Objects.equals(pointOfEntryName, that.pointOfEntryName)
+			&& Objects.equals(pointOfEntryId, that.pointOfEntryId)
+			&& disease == that.disease
+			&& Objects.equals(ageGroup, that.ageGroup);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(
+			regionName,
+			regionId,
+			districtName,
+			districtId,
+			healthFacilityName,
+			healthFacilityId,
+			pointOfEntryName,
+			pointOfEntryId,
+			year,
+			epiWeek,
+			disease,
+			newCases,
+			labConfirmations,
+			deaths,
+			ageGroup);
+	}
+
+	public boolean similar(Object o, AggregateReportGroupingLevel groupingLevel) {
```

Comment:
Why not use AggregateCaseCountDto instead of Object? This is not an equals method anyway.